cmd/fetch_repo/fetch_repo.go

```diff
@@ -13,78 +13,88 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// Command fetch_repo is similar to "go get -d" but it works even if the given
-// repository path is not a buildable Go package and it checks out a specific
-// revision rather than the latest revision.
+// Command fetch_repo downloads a Go module or repository at a specific
+// version or commit.
 //
-// The difference between fetch_repo and "git clone" or {new_,}git_repository is
-// that fetch_repo recognizes import redirection of Go and it supports other
-// version control systems than git.
+// In module mode, fetch_repo downloads a module using "go mod download",
+// verifies the contents against a sum, then copies the contents of the module
```

Comment:
I wonder if rather than copying we could just do a hardlink to save disk space? Bazel itself offers a flag for this for the repo cache: https://docs.bazel.build/versions/master/command-line-reference.html#flag--experimental_repository_cache_hardlinks & https://github.com/bazelbuild/bazel/issues/5568 for background.

Reply:
Hard links and symbolic links both cause compatibility issues. They aren't universally supported across operating systems and file systems. Windows especially has problems, and they cause problems with sandboxfs, too.

Reply:
Yeah I was thinking as much - I wonder if it is worth having some option for this? Maybe configurable via an env var, something like `BAZEL_GO_MODULE_LINK`? So that people can choose tradeoffs specifically to their setup consciously? But yeah also totally fair if you want to leave it as is for now.

Reply:
It's more a question of cost of implementation and maintenance. I'm sure it would work if we were careful, but implementing it, testing on all platforms, and fixing bugs afterward would take time. I'd rather keep it simple, at least for now.